### PR TITLE
Add strict option to lr_scheduler dict

### LINKED ISSUE #3526: Add strict option to scheduler
## 🚀 Feature

When a `ReduceLROnPlateau` scheduler is used, training stops with a raised `MisconfigurationException` if the `monitor` is not found. We should add a `strict` option (same as in the `EarlyStopping` callback) so it is configurable.

EarlyStopping:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/callbacks/early_stopping.py#L61-L62
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/callbacks/early_stopping.py#L107-L116

OptimizerConnector:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/trainer/connectors/optimizer_connector.py#L53-L59

### Motivation

If my first epoch fails to produce a valid loss, the scheduler stops training.
### Pitch

```python
    def configure_optimizers(self):
        optimizer = ...
        scheduler = {
            "scheduler": torch.optim.lr_scheduler.ReduceLROnPlateau(
                optimizer,
                mode="min",
            ),
            "monitor": "checkpoint_on",
            "interval": "epoch",
            "frequency": 1,
            "strict": False  # Allow this
        }
        return [optimizer], [scheduler]
```

### PR DESCRIPTION
**Pull request by carmocca**
Fixes #3526

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]
LGTM, test are solid and the failing checks seems to be unrelated to this PR

**Review by justusschock — APPROVED** [review]

**Comment by awaelchli** [comment]
Isn't this more like a feature/enhancement than bugfix? Is it allowed pre v1.0?

**Comment by carmocca** [comment]
Rebased master, can you review? @Borda @awaelchli @rohitgr7

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
